### DISCUSSION TITLE
Web Interface

### DISCUSSION THREAD

**Discussion by SaiPanneerselvam** [AntonOsika/gpt-engineer#505] [General]
This is the new official Web Interface discussion thread for issue #332.
View the Official Repository at [SaiPanneerselvam/gpt-engineer](https://github.com/SaiPanneerselvam/gpt-engineer)

**Comment by SumitKumarDev10** [comment]
@SaiPanneerselvam Ok, You can access my repo from [GPT-Engineer-My-Repo](https://github.com/AntonOsika/gpt-engineer).

**Reply by SaiPanneerselvam** [reply]
The Goal is no longer in that repository sadly :( 
I felt like it would be better if the repository was in the hands of the creator of the suggestion.

**Reply by SaiPanneerselvam** [reply]
With that in mind, I would like to remind you that you are allowed to be a maintainer (not contributor) in my repository.

**Reply by SaiPanneerselvam** [reply]
https://github.com/SaiPanneerselvam/gpt-engineer/invitations

**Comment by SumitKumarDev10** [comment]
My pull request is yet to be confirmed and merged into the main branch.
I have done all my work on the **`web/flask-bootstrap-python-website`** branch.
After it is forked and cloned, and once we are in the gpt-engineer directory, we can run `**python3 web/web_main.py**` command.

**Reply by SaiPanneerselvam** [reply]
not understood, but please accept the invitation to the repo.

**Reply by SumitKumarDev10** [reply]
Ok, I have accepted the invitation.

**Reply by SaiPanneerselvam** [reply]
Nice! But, how where is our project being deployed?

**Reply by SaiPanneerselvam** [reply]
i know that it is being run in the web branch but where can we see it at?

**Reply by SumitKumarDev10** [reply]
I think Vercel or Heroku is great for this task.

**Reply by SumitKumarDev10** [reply]
I have not yet deployed it yet.

**Reply by SaiPanneerselvam** [reply]
I am new to this though.

**Reply by SaiPanneerselvam** [reply]
I have only use providers like glitch, EMTIndia, IndiaGit. I don't have experience with Heroku or Vercel.

**Reply by SumitKumarDev10** [reply]
Hmmm... This is getting interesting. I will try checking out Glitch and then we can talk about how this deployment will be done.

**Reply by SaiPanneerselvam** [reply]
Yes... I agree. Maybe I will also look into Heroku and Vercel. I think you might need a subscription to use Vercel, heroku I know nothing about.

**Reply by SumitKumarDev10** [reply]
**Vercel** is fully free and **Heroku** supports maybe 1 deployment of application.

**Reply by SaiPanneerselvam** [reply]
oh, then i might have thought of it the OTHER way :)

**Reply by SaiPanneerselvam** [reply]
So, I then might be thinking of Vercel more that heroku.

**Reply by SumitKumarDev10** [reply]
Ok, let's go and explore our respective paths and will see you in a few days.

**Reply by SumitKumarDev10** [reply]
Bfn!

**Reply by SaiPanneerselvam** [reply]
I cannot use my phone number in vercel, so can we think on creating a shared account?

**Reply by SumitKumarDev10** [reply]
Let's find some other service where phone number is not required. Do you think Glitch will come to our rescue?

**Reply by SaiPanneerselvam** [reply]
I don't know. Though if you want any example website i have made they are on glitch. FYI, I did these a few years ago.
https://interestgaming.glitch.me
https://coolcraftcreate.glitch.me
https:// many more!

**Comment by SaiPanneerselvam** [comment]
(Just for fun) I just noticed that this is discussion 505, meaning SOS !!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by SumitKumarDev10** [comment]
Ohh!😯

**Comment by SaiPanneerselvam** [comment]
I know how to code in shs which is a tool to make a website compatible with ANY CODING LANGUAGE. Maybe I can use this knowledge to implement the python scripts into our project. The one problem is that the tool has python 1.2

**Reply by SumitKumarDev10** [reply]
What is this 'shs'?

**Reply by SaiPanneerselvam** [reply]
basically a tool its kind of new but they still didnt update it from python 1.2.1

**Reply by SumitKumarDev10** [reply]
Can you send me a weblink to the doc of shs? I am really unfamiliar with this term. Never heard on it before.

**Reply by SaiPanneerselvam** [reply]
So maybe we could forget about it because it ALSO does not load pypi.

**Reply by SaiPanneerselvam** [reply]
and the openai lib is neccessary

**Reply by SumitKumarDev10** [reply]
Now, I am getting really curious. I have searched it on Google but have not found a slightest hint of it.

**Reply by SaiPanneerselvam** [reply]
this tool is still in development, so there is NO official website

**Reply by SumitKumarDev10** [reply]
Okay!

**Comment by SaiPanneerselvam** [comment]
Wondering why they dont update

**Comment by SumitKumarDev10** [comment]
Hi, I am  just starting with **Glitch**.
I think we can leverage the power of **ChatGPT** in making a website for this repository.

**Reply by SaiPanneerselvam** [reply]
Yes, I agree.

**Comment by SumitKumarDev10** [comment]
I think we should create a folder named 'web' in which all the python, html and css files will be stored.

**Reply by SumitKumarDev10** [reply]
What are your thoughts on this?

**Reply by SaiPanneerselvam** [reply]
Should it be a folder or a branch? FYI, the gpt-engineer original project STILL is not getting any updates, so which repository are you on? I am using https://github.com/SaiPanneerselvam/gpt-engineer

**Reply by SumitKumarDev10** [reply]
I think of having just another folder for this.

**Reply by SaiPanneerselvam** [reply]
sure. We will just use another folder

**Comment by SumitKumarDev10** [comment]
I shall use the same link/repo as you mentioned.

**Reply by SaiPanneerselvam** [reply]
okay!

**Comment by SumitKumarDev10** [comment]
Done setting up the repo on Glitch and Local Github Desktop

**Comment by SumitKumarDev10** [comment]
Created a branch named 'web/flask-website' in the local cloned repository.
When all the commits are done, I will pull the request to the remote repo and then give a merge request.
It will take some time to complete.

**Comment by SaiPanneerselvam** [comment]
@SumitKumarDev10, please update the repository for gpt-engineer and also delete any unknown modifications, as it is making the actual repository go behind in updates.

Please understand, as our team is forward to always staying up to date.

**Reply by SaiPanneerselvam** [reply]
O yeah, I got ten other people to join.

**Reply by SumitKumarDev10** [reply]
Okay, that's great

**Comment by SaiPanneerselvam** [comment]
This discussion is getting a little stale, I am hoping the project still is alive.

**Comment by SaiPanneerselvam** [comment]
Also, when will we get api keys? I just got this idea to have apikeys but in so the user doesn't have to get confused about apikeys.

**Comment by ATheorell** [comment]
Since this effort seems to have gone stale, a new initiative has started. Anyone from this thread interested in joining forces?

**Reply by avibathula** [reply]
I am new to this initiative and was hoping to get started - but looks like it has been stale for sometime.

Kindly pull me into any new efforts spawning up.

**Reply by ATheorell** [reply]
Very cool!
Are you on our [discord](https://discord.gg/8tcDQ89Ej2)? I'm ATheorell on the discord. Reach out to me and I'll put you in contact with zdanl who is leading the web interface development.

**Comment by zdanl** [comment]
FYI https://github.com/AntonOsika/gpt-engineer/pull/860

Since we forked an open source ChatGPT template, and modified it, progress is visible. We are now wiring the new Python3 Agent code base together with a ReactJS application via Socket.io (currently still websockets). Thanks a lot to @ATheorell and sorry for being so slow here.

Any tasks on this agenda need to be done and any help would be appreciated!
https://github.com/users/zdanl/projects/10/views/1?layout=board

We realize spawning off an open source template is not very original work, but we would like to focus on intuitive implementation of required UI/UX specific to GPT Engineer, and a smooth, real-time interaction between Javascript and Python, rather than taking care mobile browser support, etc.

<img width="1768" alt="Screenshot 2023-11-22 at 20 09 29" src="https://github.com/AntonOsika/gpt-engineer/assets/114028070/93441b57-1b04-4325-b121-07a60fc3f097">

**Comment by zdanl** [comment]
Full Commit on my side. Socket.IO is now supported in the ReactJS GUI as well as in Python3 Agent Server.py

_app.tsx has 2 bugs that occured because my Javascript is ok but Typescript sucks. It's minor syntax and type definition semantics. Maybe someone can fix it? See comment.

server.py is so far ok

https://github.com/zdanl/gpt-engineer/commit/5b1bd8fa25b4b5d9ccc1dc99341c0ed577a3d8d3

**Comment by SaiPanneerselvam** [comment]
I have a confusion, what is our plan for the URL?

**Reply by zdanl** [reply]
This would be @AntonOsika decision, i just checked, gptengineer.com and gptengineer.io are taken. It would be very cool to host this!

**Reply by SaiPanneerselvam** [reply]
Maybe try gptengineer.org, as this is a free and open-source project indeed.

**Comment by SaiPanneerselvam** [comment]
Also, which repo are these changes being made in?

**Reply by zdanl** [reply]
there is a GUI branch now. the very newest before pull request is in [GUI](https://github.com/zdanl/gpt-engineer)

**Comment by zdanl** [comment]
Responses from GPT-Engineer start coming in as JSON via Socket.IO. :) API Key can be changed in web interface. 

<img width="1623" alt="Screenshot 2023-11-25 at 23 03 55" src="https://github.com/AntonOsika/gpt-engineer/assets/114028070/9243ba51-3dc2-44e1-92e6-e54e564f4b33">

**Comment by zdanl** [comment]
Video


https://github.com/AntonOsika/gpt-engineer/assets/114028070/02192f44-7a29-412a-b536-65bf7a950c49

**Comment by SaiPanneerselvam** [comment]
Another question: What to do with the API Keys, is there a way to have one global key?

**Reply by ATheorell** [reply]
Do you mean globally, but on your local computer? Different ways to do this depends a bit on your operating system. On a unix like system you can export the environment variable in something like .bashrc.

**Reply by SaiPanneerselvam** [reply]
No, my idea is that a global api key means that the api key is not required and can be used by anyone ( like [ChatTeach/FreeGPT](https://github.com/ChatTeach/FreeGPT).

**Comment by SaiPanneerselvam** [comment]
I have created a discord server to continue the discussion on. It is the link below:
[Open "Web Interface - Discussion 505" Invite](https://discord.gg/X6EXcMyMpq)

**Reply by SaiPanneerselvam** [reply]
@ATheorell I suggest that all the upcoming messages be sent in the Web Interface Discussion Discord

**Reply by ATheorell** [reply]
Sounds good! @SaiPanneerselvam

**Comment by SaiPanneerselvam** [comment]
Its amazing to see that 1 small idea started all this.

**Reply by PierrunoYT** [reply]
Discord Server is broken

**Reply by SaiPanneerselvam** [reply]
Thanks for reporting, I will look into the issue (filed as issue 1029)

**Reply by PierrunoYT** [reply]
How can I test the GUI?

**Comment by PierrunoYT** [comment]
How can I test the UI. I downloaded the Develope Release as the other one not working like the dev release.
